test/base_test_fixture.h

```diff
@@ -1502,6 +1502,38 @@ struct base_test_fixture
         results.get_ref(0, ref);
         REQUIRE(ref == name);
     }
+    void string_vector_test()
+    {
+        nanodbc::connection connection = connect();
+        REQUIRE(connection.native_dbc_handle() != nullptr);
+        REQUIRE(connection.native_env_handle() != nullptr);
+        REQUIRE(connection.transactions() == std::size_t(0));
+
+        const std::vector<nanodbc::string_type> first_name = {NANODBC_TEXT("Fred"), NANODBC_TEXT("Barney"), NANODBC_TEXT("Dino")};
+        const std::vector<nanodbc::string_type> last_name = {NANODBC_TEXT("Flintstone"), NANODBC_TEXT("Rubble"), NANODBC_TEXT("")};
+
+        drop_table(connection, NANODBC_TEXT("string_test"));
+        execute(connection, NANODBC_TEXT("create table string_test (first varchar(10), last varchar(10));"));
+
+        nanodbc::statement query(connection);
+        prepare(query, NANODBC_TEXT("insert into string_test(s) values(?, ?)"));
```

Comment:
It looks to me this test is incomplete.

1. It is not called anywhere.
2. SQL statements use incorrect number of columns.
3. `bind_strings` is called with incorrect column : `0` and `2`.
4. Table name `string_test` is used by the other test case, it's better to keep unique table names per test case.

I have taken the liberty to update the tests as follows:

```
diff --git a/test/base_test_fixture.h b/test/base_test_fixture.h
index 311656a..b87df04 100644
--- a/test/base_test_fixture.h
+++ b/test/base_test_fixture.h
@@ -1512,18 +1512,18 @@ struct base_test_fixture
         const std::vector<nanodbc::string_type> first_name = {NANODBC_TEXT("Fred"), NANODBC_TEXT("Barney"), NANODBC_TEXT("Dino")};
         const std::vector<nanodbc::string_type> last_name = {NANODBC_TEXT("Flintstone"), NANODBC_TEXT("Rubble"), NANODBC_TEXT("")};
 
-        drop_table(connection, NANODBC_TEXT("string_test"));
-        execute(connection, NANODBC_TEXT("create table string_test (first varchar(10), last varchar(10));"));
+        drop_table(connection, NANODBC_TEXT("string_vector_test"));
+        execute(connection, NANODBC_TEXT("create table string_vector_test (first varchar(10), last varchar(10));"));
 
         nanodbc::statement query(connection);
-        prepare(query, NANODBC_TEXT("insert into string_test(s) values(?, ?)"));
+        prepare(query, NANODBC_TEXT("insert into string_vector_test(first, last) values(?, ?)"));
         query.bind_strings(0, first_name, 3);
 
         // With null sentry
-        query.bind_strings(2, last_name, 3, NANODBC_TEXT(""));
+        query.bind_strings(1, last_name, 3, NANODBC_TEXT(""));
         nanodbc::execute(query);
 
-        nanodbc::result results = execute(connection, NANODBC_TEXT("select s from string_test;"));
+        nanodbc::result results = execute(connection, NANODBC_TEXT("select first,last from string_vector_test order by first desc;"));
         REQUIRE(results.next());
         REQUIRE(results.get<nanodbc::string_type>(0) == NANODBC_TEXT("Fred"));
         REQUIRE(results.get<nanodbc::string_type>(1) == NANODBC_TEXT("Flintstone"));
diff --git a/test/mssql_test.cpp b/test/mssql_test.cpp
index 12af1a1..2c5d40f 100644
--- a/test/mssql_test.cpp
+++ b/test/mssql_test.cpp
@@ -355,6 +355,11 @@ TEST_CASE_METHOD(mssql_fixture, "string_test", "[mssql][string]")
     string_test();
 }
 
+TEST_CASE_METHOD(mssql_fixture, "string_vector_test", "[mssql][string]")
+{
+    string_vector_test();
+}
+
 TEST_CASE_METHOD(mssql_fixture, "transaction_test", "[mssql][transaction]")
 {
     transaction_test();
```

I tried it against SQL Server 2016 and there seem to be a problem:
* Only first elements from the bound vectors are inserted

```
SELECT [first], [last] FROM [string_vector_test]

first | last
-------------
Fred | Flintstone
```